ionic RC1

WoH opened the pull request:
![](https://media.giphy.com/media/rl0FOxdz7CcxO/giphy.gif)

See: https://github.com/PokemonGoers/Catch-em-all/issues/90

**Big thanks to @johartl for a lot of his work on [Catch-em-all/tree/rc0](https://github.com/PokemonGoers/Catch-em-all/tree/rc0)**

WoH committed:
package.json Bump to RC1

WoH committed:
[RC1] New structure

WoH committed:
[RC1] Environments

WoH committed:
[RC1] .gitignore

WoH committed:
[RC1] Themes

WoH committed:
[RC1] Component headers

WoH committed:
[RC1] RXJS isUnsubscribed -> closed

WoH committed:
[RC1] TS imports

WoH committed:
[RC1] Rollup: Recover from refactor

WoH committed:
[RC1] Package.json: ES6 Shim types
Object.assign, Promises etc

WoH committed:
[RC1] Component header: styleUrls

WoH committed:
[RC1] No tests, no typings

WoH committed:
[RC1] AppModule: Add EvolutionsComponent

WoH committed:
[RC1] ConfigService: Platform

WoH committed:
[RC1] Add Default ServiceWorker

WoH committed:
[RC1] Component: Ionic styles

WoH committed:
WikiIndex: Missing method onCancel()

WoH committed:
TypesComponent: Declare types

WoH committed:
POICardComponent: Use property binding for @slide

WoH committed:
Navbar: title input property

WoH committed:
Search Page: Use NavbarComponent

WoH committed:
PWA: Add manifest and serviceworker stub

WoH committed:
[RC1] Rollup Config: env location

WoH committed:
[RC1] ConfigService / env fixes

WoH committed:
[RC1] Typescript Config

WoH committed:
[RC1] Fix Injector

WoH committed:
[RC1] AppComponent: DI and template

WoH committed:
[RC1] Remove es6-shim types

WoH committed:
FilterPokemonTab: Fix type-definition Access

WoH committed:
[RC1] Navbar: ion-button

WoH committed:
[RC1] Page-Selectors

WoH committed:
[RC1] Missing aout Page Selector

WoH committed:
[RC1] Scope styles

WoH committed:
[RC1] new asset location

WoH committed:
[RC1] Nav-Sidebar: Hack to show list below title

WoH committed:
[RC1] Popover: Fix all the things
Mostly implementing new button / fab stuff and new scss style
Remove old ChangeDetection Hacks, god bless

WoH committed:
[RC1] Searchbars: Initialize with empty input string
Otherwise, undefined will be shown initially

WoH committed:
[RC1] Rollup: Add Websocket Endpoint

WoH committed:
[RC1] Popover: Missing Methods, unused imports

WoH committed:
[RC1] Travis: Use ionic-app-scripts

WoH committed:
[RC1] New button and coloring behavior continued

WoH committed:
[RC1] New lifecycle events

WoH committed:
[RC1] Splash screen location

WoH committed:
[RC1] Evolutions: Fix Css

WoH committed:
Filter: width

WoH committed:
[RC1] Hack: pre-defined endpoints, overwrite in prod

WoH committed:
[RC1] Travis: Builds in proper directory

WoH committed:
Merge branch 'develop' into rc1

WoH committed:
[RC1] New button styles

WoH committed:
Unused imports

WoH committed:
[RC1] Expose Config for AoT compiling

WoH committed:
[RC1] icon-buttons

WoH committed:
[RC1] Force release env

WoH committed:
Update Readme

WoH committed:
[RC1] Refactoring derp

AlexanderLill commented:
Wow, that is ... a _lot_.

@sacdallago and @gyachdav should get their seminar statistics right - before you basically remove all our code ... ;)

Will review this tomorrow, wanna do this in one go before falling asleep...

WoH commented:
I left everything in tact whenever possibe. I moved a lot of the stuff from /app... to /src, but I also had to scope the styles differently and as soon as that happens, github doesn't count it as moving a -> b but as removing a and adding b.

Also, when did statistics begin to matter? Just slip the build folder into one of your commits, add one to remove it and merge it before I can retaliate.

MajorBreakfast commented:
First: Good work! But...

![3875822](https://cloud.githubusercontent.com/assets/340142/19549748/60ac4196-96a6-11e6-9ccd-986e3429a01e.jpg)

We're loosing scoped styles?

sacdallago commented:
Hey guys will this PR fix the `ERROR NativeStorageError`? It's like super annoying!

I guess that's why the map is not loading? Maybe it also has to do with maxing out the requests at mapbox.

Just please merge if this is fine and fixes issues, we are like super late on schedule.

WoH committed:
Use the NativeStorage only after platform is ready

WoH commented:
@sacdallago thanks for reminding me of that. However, this is not related to RC1 and this PR is already not the smallest. I opened another one that tackles that issue. Once that PR is merged and #136 is solved, the map will load and load without errors.

AlexanderLill commented on .travis.yml:
```diff
@@ -47,11 +46,11 @@ script:
   # Run frontend/app tests
   - cd ../ionic2
-  - npm test
+  # - npm test
```
I guess this is removed because we do not have any tests?

WoH replied:
Exactly

AlexanderLill reviewed (changes requested):
Looks good to me :)

sacdallago commented:
Can this finally be merged?

WoH commented:
Waiting for @johartl to confirm or add his thoughts.

johartl commented:
I will have a look today. Sorry for the delay.

WoH committed:
Merge branch 'develop' of github.com:PokemonGoers/Catch-em-all into rc1

WoH committed:
[RC1] Enable ServiceWorker

Commit:
Fix environment variable exports

WoH commented:
You need both I think, I was just about to fix that :D

WoH committed:
Merge branch 'rc1' of github.com:PokemonGoers/Catch-em-all into rc1

WoH committed:
[RC1] RollupCOnfig

WoH committed:
[RC1] Reenable Proxying

WoH committed:
[RC1] ServiceWorker: Disable logging of network requests

WoH commented:
@johartl Is this good to go? I re-enabled everything that I had to work around and I think this is ready now.

Edit: The agressive caching of the ServiceWorker should also massively decrease the data consumption.

johartl commented:
Please give me some more time. Currently the Docker build is not working and I also want to test the Android build before merging it.

WoH commented:
@sacdallago is coming after us, I agree we shouldn't make any overzealous moves though.

WoH committed:
[RC1} Use recommended Angular version

WoH committed:
[RC1] move theme/global.scss to app/app.scss
https://github.com/driftyco/ionic2-app-base/commit/6bce14a580878745a80032da0c90715cda047952

Commit:
Fix docker build

sacdallago commented:
@WoH you bet I am

johartl commented:
@sacdallago I'm sure the last thing you want is a broken Docker build though ;)
From my side this PR is good to go now!

sacdallago commented:
@johartl oh, well, you know how to make a man happy, don't you?

WoH commented:
**^**
![KappaPride](https://static-cdn.jtvnw.net/emoticons/v1/55338/3.0)

I will merge as soon as I can conform that it integrates nicely with PM-1

WoH committed:
Use latest Poke/Map

sacdallago commented:
oh wow, this happened
